core/src/main/scala/com/softwaremill/events/EventMachine.scala

```diff
@@ -45,6 +46,24 @@ class EventMachine(
       })
   }
 
+  def failOverFromStoredEventsTo(timeLimit: OffsetDateTime = OffsetDateTime.now()): Unit = {
+    storedEventsCount().foreach(eventsCount => logger.info(s"Number of events to recover: $eventsCount"))
```

Comment:
In theory it's possible that this will be logged after the whole recovery ;). Maybe use ` for { <- ` to combine these futures?